crates/cairo-profiler/src/profile_builder.rs

```diff
@@ -267,3 +276,27 @@ pub fn build_profile(samples: &[Sample]) -> pprof::Profile {
         default_sample_type: 0,
     }
 }
+
+pub fn save_profile(target_path: &Utf8PathBuf, profile: &pprof::Profile) -> Result<()> {
+    if let Some(parent) = target_path.parent() {
+        fs::create_dir_all(parent)
+            .context("Failed to create parent directories for the output file")?;
+    }
+    let mut file = fs::File::create(target_path).context("Failed to create the output file")?;
+
+    let mut buffer = BytesMut::new();
+    profile
+        .encode(&mut buffer)
+        .expect("Failed to encode the profile to the buffer");
+
+    let mut buffer_reader = buffer.reader();
+    let mut encoder = GzEncoder::new(&mut buffer_reader, Compression::default());
+
+    let mut encoded_buffer = vec![];
+    encoder
+        .read_to_end(&mut encoded_buffer)
+        .context("Failed to read bytes from the encoder")?;
+    file.write_all(&encoded_buffer).unwrap();
```

Comment:
[super nit]
```suggestion
            fs::write(&build_cli.output_path, &encoded)?;
```

This way we don't need 

```
let mut file = fs::File::create(target_path).context("Failed to create the output file")?;
```